### PR TITLE
ASR: Adding new Resource replication protection intent and its associated PUT and GET call

### PR DESCRIPTION
**Pull request by ayfathim**
We are planning to integrate ASR as part of the create VM Blade.
For this, we are introducing a new resource namely “ReplicationProtectionIntent”. And we have added GET, PUT and List API for this resource.

During create vm flow, the  replicationProtectionIntent object is created for the VM, the ASR service will take care of enabling replication for the VM internally.
This new PUT API  for replicationProtectionIntent is a synchronous API.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ayfathim** [comment]
@Azure/arm-api-review-board : Please review this. We need this for ignite demo, hence request you to please prioritize

**Comment by ayfathim** [comment]
I have raised request for suppressing the error https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r4010 , as per guidelines defined here: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process?anchor=obtaining-suppression

**Review by ravbhatnagar — COMMENTED** [review]
a couple of comments to fix, but approving hoping that those comments will be trivial for you to address.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 1]
```diff
@@ -4070,6 +4070,153 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets the list of replication protection intent objects.": {
+            "$ref": "./examples/ReplicationProtectionIntents_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationProtectionIntents/{intentObjectName}": {
```
[the remark this answers is not in the document]
Comment (ayfathim) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:4116]: Delete API for protection Intent is not exposed to user.

**Comment by qianwens** [comment]
@ayfathim please help fix the linter error to get this PR qualified to merge.
The response is defined but without a default error response implementation.Consider adding it.'
  at specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:4098

**Comment by ayfathim** [comment]
@qianwens : Our service currently does not have the default error response defined in the format that is needed. i have opened a workitem to do this post ignite: https://msazure.visualstudio.com/One/_workitems/edit/8045859/
Let me know, if i can byPass this validation check for now.

**Comment by qianwens** [comment]
> @qianwens : Our service currently does not have the default error response defined in the format that is needed. i have opened a workitem to do this post ignite: https://msazure.visualstudio.com/One/_workitems/edit/8045859/
> Let me know, if i can byPass this validation check for now.

You need to follow this process to suppress the rule https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process

**Comment by ayfathim** [comment]
@qianwens : I have raised swagger exception for the default error response. And as its ARM signed off, can i proceed ahead with adding the suppression code for default error response?
image

**Comment by ayfathim** [comment]
https://github.com/Azure/openapi-diff/blob/master/docs/rules/1038.md
The "AddedPath" looks like a false positive to me. I have not added any subResource under a resource.
@qianwens : Can you please help me understand, why this is failing?

**Comment by ayfathim** [comment]
I have added suppression for the "default error response" error, as the ARM team has approved it.
image

**Comment by qianwens** [comment]
> https://github.com/Azure/openapi-diff/blob/master/docs/rules/1038.md
> The "AddedPath" looks like a false positive to me. I have not added any subResource under a resource.
> @qianwens : Can you please help me understand, why this is failing?

You added new path /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationProtectionIntents/{intentObjectName} to existing path /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}. so that you should add a new API version

**Comment by qianwens** [comment]
[3:56 PM] Ayesha Fathima
    it is created by customer....but the call goes to different service "recovery services"................"recovery services" will inform ASR about vault created......and the user can protect a vm inside the vault by point a PUT call, which will create a replication protection intent resource
​[3:56 PM] Ayesha Fathima
    none of the properties of vault resource is impacted by the new resources being added "replicationProtectionIntent"
Edited​[3:59 PM] Ayesha Fathima
    its a breaking change for the vault resource.....its independent of it.

**Review by qianwens — APPROVED** [review]

**Pull request merged**